src/_tkagg.cpp

```diff
@@ -265,20 +264,20 @@ int get_tk(HMODULE hMod)
     if (TK_MAIN_WINDOW == NULL) {  // Maybe not Tk module
         return 0;
     }
-    return ( // -1 if any remaining symbols are NULL
+    return  // -1 if any remaining symbols are NULL
         ((TK_FIND_PHOTO = (Tk_FindPhoto_t)
```

Comment:
I think it would be clearer if these were a bunch of separate `if`s instead of this long chained thing (if every assignment fit on one line, that would also be clearer.)

Reply:
That'll be much simpler to do after #10680 is merged and we later get to remove the old version of the API.  Then the only functions we'll have to load are Tk_FindPhoto and Tk_PhotoPutBlock_NoComposite, which also means that the Windows and non-Windows versions will only need to differ by how they call _dfunc.
Not convinced it's worth changing for now.

Reply:
Just to be clear, the endgame is to just do (e.g.)
```
void *_dfunc(void *lib_handle, const char *func_name);

int _func_loader(void *lib)
{
    // Fill global function pointers from dynamic lib.
    // Return number of successfully loaded functions.
    #define LOAD_TK(name) tk::name = Tk_##name##_t(_dfunc(lib, "Tk_" #name))
    return bool(LOAD_TK(FindPhoto)) + bool(LOAD_TK(PhotoPutBlock_NoComposite));
    #undef LOAD_TK
}
```
followed by OS-specific versions of `_dfunc`.

Reply:
#10680 is merged now.

Reply:
But the old API is still around (the simplification will only come after `matplotlib.backends.tkagg` is completely removed).